core/__tests__/modules/codeConfig/codeConfig.ts

```diff
@@ -577,25 +586,267 @@ describe("modules/codeConfig", () => {
       expect(sources[0].id).toBe("users_table");
       expect(sources[0].state).toBe("deleted");
       expect(sources[0].locked).toBeNull();
-
-      await specHelper.runTask("source:destroy", { sourceId: "users_table" });
     });
 
     test("a removed app will be deleted", async () => {
-      const apps = await App.scope(null).findAll();
-      expect(apps.length).toBe(1);
-      expect(apps[0].id).toBe("data_warehouse");
-      expect(apps[0].state).toBe("deleted");
-      expect(apps[0].locked).toBeNull();
-
-      await specHelper.runTask("app:destroy", { appId: "data_warehouse" });
+      const app = await App.scope(null).findOne({
+        where: { id: "data_warehouse" },
+      });
+      expect(app.state).toBe("deleted");
+      expect(app.locked).toBeNull();
     });
 
     test("settings remain", async () => {
       expect(await Setting.count()).toBeGreaterThan(1);
     });
   });
 
+  describe("bring it all back", () => {
+    let previousRun;
```

Comment:
It might be good to have a run for a few types of things that got deleted, in addition to the destination - App, Property, etc.